.github/workflows/performance.yml

```diff
@@ -0,0 +1,56 @@
+name: Performance Benchmark
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  # Check tests pass on multiple Python and OS combinations
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macOS-latest, windows-latest]
```

Comment:
Is this the full matrix with python versions? do we want a full matrix?

Reply:
Nah just the latest - wasn't sure if we wanted all the python versions as opposed to just the latest since we aren't checking for failures as opposed to just speed. IMO it doesn't seem necessary. However, the OS seems highly relevant to the speed since it might change things like memory and core availability

Reply:
I agree, we definitely want OS. Most recent sounds good. I could see an issue where one of our dependencies has varying performance at different python versions but that's probably too fringe